Blog-projects/Widget-window-in-Qt-Quick-app/fontcontrolswidgetsform.cpp

```diff
@@ -0,0 +1,47 @@
+#include "fontcontrolswidgetsform.h"
+#include "ui_fontcontrolswidgetsform.h"
+
+FontControlsWidgetsForm::FontControlsWidgetsForm(QWidget *parent)
+    : QWidget(parent)
+    , ui(new Ui::FontControlsWidgetsForm)
+{
+    ui->setupUi(this);
+    connect(ui->lineEdit, &QLineEdit::textEdited, this, &FontControlsWidgetsForm::lineEdit_textChanged);
+    connect(ui->fontComboBox, &QFontComboBox::currentFontChanged, this, &FontControlsWidgetsForm::fontComboBox_currentFontChanged);
+    connect(ui->pushButton, &QPushButton::clicked, this, &FontControlsWidgetsForm::pushButton_clicked);
+}
+
+FontControlsWidgetsForm::~FontControlsWidgetsForm()
+{
+    delete ui;
+}
+
+const QString FontControlsWidgetsForm::text()
+{
+    return ui->lineEdit->text();
+}
+
+void FontControlsWidgetsForm::lineEdit_textChanged(const QString &arg1)
+{
+    emit textChanged();
+}
+
+void FontControlsWidgetsForm::setText(const QString &text)
+{
+    ui->lineEdit->setText(text);
+    emit textChanged();
+}
+
+const QString FontControlsWidgetsForm::font()
```

Comment:
Misunderstanding.
Returning a const QString serves no purpose (on the contrary - it prevents move semantics for the caller)

When I said "make those methods const", I meant marking the actual methods themselves as const:

QString FontControlsWidgetsForm::font() const

(both in .h and .cpp)

(applies to *all* getters)

The purpose is to indicate (and ensure) that the member variables are not (and cannot be) modified in such methods, because in a const method, all member variables are const.